java/flight/flight-core/src/test/java/org/apache/arrow/flight/TestCallOptions.java

```diff
@@ -64,6 +67,61 @@ public void underTimeout() {
     });
   }
 
+  @Test
+  public void singleProperty() {
+    final MetadataAdapter headers = new MetadataAdapter(new Metadata());
```

Comment:
We probably need a CallHeaders implementation that doesn't come from the flight-grpc package, or there's no way to use HeaderCallOption.